ownCloudAppShared/Client/Sharing/ShareViewController.swift

```diff
@@ -250,26 +261,22 @@ open class ShareViewController: CollectionViewController, SearchViewControllerDe
 		bottomButtonBarViewController.view = bottomButtonBar
 
 		// - Add delete button for existing shares
-		if mode == .edit, let bottomButtonBar {
+		if mode == .edit {
 			var deleteButtonConfig = UIButton.Configuration.borderedProminent()
 			deleteButtonConfig.title = "Unshare".localized
 			deleteButtonConfig.cornerStyle = .large
 			deleteButtonConfig.baseBackgroundColor = .systemRed
 			deleteButtonConfig.baseForegroundColor = .white
-
+			
 			let deleteButton = UIButton()
-			deleteButton.translatesAutoresizingMaskIntoConstraints = false
 			deleteButton.configuration = deleteButtonConfig
 			deleteButton.addAction(UIAction(handler: { [weak self] action in
 				self?.deleteShare()
 			}), for: .primaryActionTriggered)
-
-			bottomButtonBar.addSubview(deleteButton)
-
-			NSLayoutConstraint.activate([
-				deleteButton.leadingAnchor.constraint(equalTo: bottomButtonBar.leadingAnchor, constant: 20),
-				deleteButton.centerYAnchor.constraint(equalTo: bottomButtonBar.selectButton.centerYAnchor)
-			])
+			
+			let unshare = UIBarButtonItem(customView: deleteButton)
+			
+			self.navigationItem.rightBarButtonItem = unshare
```

Comment:
Just my $0.02: I think it would fit and look better to have this button in a non-bordered, non-backgrounded, destructive (red) style - maybe as actual `UIBarButtonItem` button.